apps/data/src/lib/enrich/enrich/macro_execution.py

```
@@ -0,0 +1,134 @@
+        except BackendIntegrationError as e:
+            # Mark all items in this batch as failed
+            for idx in idx_map:
+                errors[idx] = f"Backend API error: {str(e)}"
+            return pd.DataFrame({"result": results, "error": errors})
+
+        # Map results back by item id
+        result_by_id = {}
+        for r in response.get("results", []):
+            result_by_id[r["id"]] = r
+
+        for item, df_idx in zip(items, idx_map):
```

Ruff (0.15.10)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)